File: ios/Src/ReactNative/RNINavigatorRouteHeaderView/RNINavigatorRouteHeaderViewManager.m
//
//  RNINavigatorRouteHeaderViewManager.m
//  react-native-ios-navigator
//
//  Created by Dominic Go on 4/14/21.
//

#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(RNINavigatorRouteHeaderViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(config, NSDictionary);
RCT_EXPORT_VIEW_PROPERTY(headerTopPadding, NSString);

@end

File: ios/Src/ReactNative/RNINavigatorRouteHeaderView/RNINavigatorRouteHeaderViewManager.swift
//
//  RNINavigatorRouteHeaderViewManager.swift
//  react-native-ios-navigator
//
//  Created by Dominic Go on 4/14/21.
//

import Foundation

@objc(RNINavigatorRouteHeaderViewManager)
internal class RNINavigatorRouteHeaderViewManager: RCTViewManager, RCTInvalidating {
  
  static var sharedBridge: RCTBridge?;
  
  // MARK: - RN Module Setup
  // ----------------------
  
  override static func requiresMainQueueSetup() -> Bool {
    // run init in bg thread
    return false;
  };
  
  override func view() -> UIView! {
    // save a ref to this module's RN bridge instance
    if Self.sharedBridge == nil {
      Self.sharedBridge = self.bridge;
    };
    
    return RNINavigatorRouteHeaderView(bridge: self.bridge);
  };
  
  override func shadowView() -> RCTShadowView! {
    return RNINavigatorRouteHeaderShadowView();
  };
  
  @objc func invalidate(){
    /// reset ref to RCTBridge instance
    Self.sharedBridge = nil;
  };
};
